packages/edit-site/src/components/editor/index.js

```diff
@@ -18,33 +24,44 @@ import Header from '../header';
 import Sidebar from '../sidebar';
 import BlockEditor from '../block-editor';
 
-function Editor( { settings } ) {
+const Context = createContext();
+export function useEditorContext() {
+	return useContext( Context );
+}
+
+function Editor( { settings: _settings } ) {
+	const [ settings, setSettings ] = useState( _settings );
 	const template = useSelect(
 		( select ) =>
 			select( 'core' ).getEntityRecord(
 				'postType',
-				'wp_template',
+				settings.templateType,
 				settings.templateId
 			),
-		[]
+		[ settings.templateType, settings.templateId ]
 	);
+	const context = useMemo( () => ( { settings, setSettings } ), [
```

Comment:
Should we use a store instead to match the pattern used in all other modules `core/edit-post` `core/editor`...?

Reply:
I think that's a bit overkill right now. We can see if we need it later, but I did it this way to avoid boilerplate.

Reply:
I'm afraid if we don't do it properly now, we'll forget about it and keep adding adhoc state. I feel we should do it sonner rather than alter.